pysd/py_backend/utils.py

```diff
@@ -409,4 +517,79 @@ def get_value_by_insensitive_key_or_value(key, dict):
         if real_value.lower() == lower_key:
             return real_value
 
-    return None
\ No newline at end of file
+    return None
+
+
+def rearrange(data, dims, coords):
+    """
+    Returns a xarray.DataArray object with the given coords and dims
+
+    Parameters
+    ---------
+      data: float or xarray.DataArray
+        The input data to rearrange.
+      dims: list
+        Ordered list of the dimensions.
+      coords: dict
+        Dictionary of the dimension names as a keys with their values.
+      switch: bool
+        Flag to denote if the dimensions can be switched. Default True,
+        The False is used to rearrange general expressions
+
+    Returns
+    -------
+      xarray.DataArray
+
+    """
+    # subset used coords in general coords will be the subscript_dict
+    coords = {dim: coords[dim] for dim in dims}
+    if isinstance(data, xr.DataArray):
+        if data.shape == tuple(compute_shape(coords, dims)):
+            # Allows switching dimensions names and transpositions
+            return xr.DataArray(data=data.values, coords=coords, dims=dims)
+
+        # The coordinates are expanded or transposed
+        # TODO replace cleaner version for Python 3 (when deprecate Py2)
+        # return xr.DataArray(0, coords, dims)
+        return xr.DataArray(np.zeros(compute_shape(coords, dims)),
+                            coords, dims) + data
+
+    else:
+        # TODO replace cleaner version for Python 3 (when deprecate Py2)
+        # return xr.DataArray(float(data), coords, dims)
+        return xr.DataArray(np.full(compute_shape(coords, dims),
+                                    float(data)), coords, dims)
+
+
+def round(x):
+    """
+    Redefinition of round function with try except
+    """
+    try:
+        return round(x)
+    except Exception:
+        return x
+
+
+def add_entries_underscore(*dictionaries):
```

Comment:
I assume this is used in the subscript names? Can you give a description of why it needs to do this?

Reply:
This is something really weird that Vensim lets using expressions like
```
GET_XLS_DATA(...)
IF_THEN_ELSE(...)
```
this function just takes the dictionaries from build and add the entry `if_then_else` pointing to the `if then else` value to make the expressions above work.

Reply:
got it.